hbase-client/src/main/java/org/apache/hadoop/hbase/client/PerClientRandomNonceGenerator.java

```diff
@@ -32,13 +32,14 @@
 public final class PerClientRandomNonceGenerator implements NonceGenerator {
 
   private static final PerClientRandomNonceGenerator INST = new PerClientRandomNonceGenerator();
-
-  private final Random rdm = new Random();
   private final long clientId;
 
+  private Random rng;
```

Comment:
I do not see any real differences here? What is the spotbugs issue here?

Reply:
Spotbugs wants you to allocate a Random object and reuse it as often as possible. This is because if you do not reuse Random, if you just do like so in a method

    Random r = new Random();
    // do something with 'r'

the quality of the generated random numbers is poor.

Reply:
I would say let's just use ThreadLocalRandom, unless SecureRandom is required, then we could create a static SecureRandom instance and use it.